packages/fungible-token/bridge-fungible-token/implementation/src/main.sw

```diff
@@ -169,8 +170,9 @@ impl Bridge for Contract {
         });
     }
 
+    #[storage(read)]
     fn bridged_token_gateway() -> b256 {
-        BRIDGED_TOKEN_GATEWAY
+        storage.l1_gateway.try_read().unwrap_or(BRIDGED_TOKEN_GATEWAY)
```

Comment:
Why this was moved to be a storage?

Reply:
I cannot recall if this was Swayam or me, but this is not set in stone yet.

Depending on how you initialize the contract's parameters, you might want to have these in configurables (constants) or in storage. You basically do not want your contract to be callable during the initialisation stage.

In that sense I think we are just following EVM mechanics here. We will have to review the whole design once it's finished. Especially because the final proxy to implement was the simple one, not diamonds - which is how I began this PR.

Fwiw I think we are going to move this one back to configurables. Again, nothing is final yet.

Reply:
Removed